api/services/SiteMembershipCreator.js

```diff
@@ -52,8 +52,15 @@ const createSiteMembership = ({ user, siteParams }) => {
   ).then(() => site);
 };
 
-const revokeSiteMembership = ({ user, site, userId }) =>
-  GitHub.checkPermissions(user, site.owner, site.repository)
+const revokeSiteMembership = ({ user, site, userId }) => {
+  if (user.username.toLowerCase() === site.owner.toLowerCase()) {
```

Comment:
I think this check is incorrect. It is essentially blocking the repo owner from revoking access of any other user.

I think it needs to be something like

```js
if (userId == user.id && user.username.toLowerCase() === site.owner.toLowerCase()) {
   // ...
}
```

Reply:
hmm ok, I'll have to check the tests again then, they all pass but this behavior is obviously not correct.